Layout fixes.

kattni opened the pull request:
<!--- Describe your changes in detail -->
I've fixed the following in this PR:
- Code colors on dark mode blended in with the background.
- Code colors on the dark version of auto-mode did the same.
- The center column was inheriting width incorrectly.
- The font in the center and right columns was decreased from original.
- The "Availability" caption was shifted to below the charts.
- The footer layout was broken.
- Headers included unnecessary margins, and spacing above was insufficient.
<!--- What problem does this change solve? -->
<!--- If this PR relates to an issue, include Refs #XXX or Fixes #XXX -->
Fixes #3 
Fixes #4 
Fixes #5 

## PR Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

kattni commented:
> Summary of what I'm seeing:
> 
> * The article body _width_ is now unchanged from the "stable" version; font sizes in the article have also been restored. However, font sizes on the sidebar seem a little larger,
> 
> * The sidebar font also seems to impact on the position of the sidebar, so the whole document is shifted to the right a little.
> 
> * The sidebar on the left and the TOC bar on the right are also _slightly_ thinner than the "stable" version

Let me know how and if you want me to address any of the bits that aren't addressed below.

> * Inline `code` blocks are rendering with red text:
> 
> Previously it was 'default' coloured:

Set this and the rest of the code font changes to the default variable.
 
> Regarding my comments on #5 - my initial thought was that we just had some over-enthusiastic rules overruling Furo styling; after a closer look, I can see the issue is more one of Bootstrap default fighting Furo defaults. If removing Bootstrap completely isn't an option (and I get the impression it would be possible, but probably more effort than it's worth), then style tweaks in the mode you've done here seem like the best approach.

This was my thought on it as well, and the reason I took this approach.

> In terms of what is an "acceptable outcome" - I don't think 100% pixel-perfect reversion to old layout sizes is necessary as a goal. I'd even argue that the increase in font size in the sidebar, while accidental, is an improvement. However, I do think we should try to veer a little more towards the original "stable" width and left/right-margin when it comes to the sidebars - in the current rendering, the sidebar is about the same width as the left margin:
> 
> Whereas previously, the sidebar is about twice the width of the margin:
> 
> I think the former spacing makes better use of the available space.

Font size was responsible for this. Fixed.

freakboy3742 reviewed (approved):
I'll flag @mhsmith for a review since he noticed the original discrepancies; but to me, this looks great.

mhsmith commented on src/beeware_theme/_static/beeware-theme.css:
```diff
@@ -11,11 +11,23 @@ body[data-theme="light"] {
 }
 
-/* Fixes issue with background colors on auto color mode */
+/* Fixes issue with code font colors on dark mode */
```
Say what the issue is; nobody will remember a month from now. Likewise for the similar comment below.

mhsmith commented on src/beeware_theme/_static/beeware-theme.css:
```diff
@@ -34,6 +46,19 @@ body[data-theme="light"] {
+}
+
+.main {
+    box-sizing: unset;
```
It's possible to guess what most of the other rules are for, but this one should have a comment explaining it.

mhsmith commented:
> The font in the center and right columns was decreased from original.

As I mentioned in https://github.com/beeware/beeware-theme/issues/4#issuecomment-2991723217, Furo's font sizes are dependent on the window width. I think this is a mistake (making the window bigger should never reduce the amount of visible text), and I'd prefer to always use the smaller font size for the body, and (like @freakboy3742 above), the larger font size for the right sidebar. This was done in the Chaquopy docs as follows:
* https://github.com/chaquo/chaquopy/blob/16.1.0/product/runtime/docs/sphinx/_static/chaquopy.css#L3
* https://github.com/chaquo/chaquopy/blob/16.1.0/product/runtime/docs/sphinx/conf.py#L107

Can we also extend the theme demo site so it can be used to verify the following?
* Code colors
* Table caption position
* Left sidebar and footer (these both require the site to have more than one page to display the navigation elements)

kattni commented:
> As I mentioned in [#4 (comment)](https://github.com/beeware/beeware-theme/issues/4#issuecomment-2991723217), Furo's font sizes are dependent on the window width. I think this is a mistake (making the window bigger should never reduce the amount of visible text), and I'd prefer to always use the smaller font size for the body, and (like @freakboy3742 above), the larger font size for the right sidebar. This was done in the Chaquopy docs as follows:
> 
>     * https://github.com/chaquo/chaquopy/blob/16.1.0/product/runtime/docs/sphinx/_static/chaquopy.css#L3
> 
>     * https://github.com/chaquo/chaquopy/blob/16.1.0/product/runtime/docs/sphinx/conf.py#L107

I have made the requested changes, and added more informative comments.
 
> Can we also extend the theme demo site so it can be used to verify the following?
> 
>     * Code colors
> 
>     * Table caption position
> 
>     * Left sidebar and footer (these both require the site to have more than one page to display the navigation elements)

I didn't create the demo site, but I can try to add this. I am unsure whether you require this to test this PR, or if this is better suited to a separate PR. Please let me know which you prefer.

kattni commented:
Added:
- Inline code for color verification.
- Tabbed content with code for color verification. (Required adding tabs plugin.)
- Table with title caption for testing caption placement.
- Multiple sections, one with with multiple pages for testing navigation and left sidebar.

Updated:
- I remember fixing the title placement, but it was not present, so I imagine I lost it somewhere in testing. Fixed.

mhsmith commented:
Looks good, just a couple more comments. Compared to Furo:

* The table caption has changed from black to grey. (The size and weight of the Toga captions were in the Toga-specific style sheet, so they don't need to be dealt with here.)
* The spare space on the left and right is no longer equally distributed:

<img width="1458" alt="Screenshot 2025-06-24 at 20 18 32" src="https://github.com/user-attachments/assets/e0958334-f34f-4584-a189-d3e35d0d87aa" />

---

<img width="1458" alt="Screenshot 2025-06-24 at 20 18 36" src="https://github.com/user-attachments/assets/e40ef342-6179-406a-bc8b-562baf886223" />

kattni commented:
> The table caption has changed from black to grey. (The size and weight of the Toga "Availability" captions are set by the Toga-specific style sheet, so they don't need to be dealt with here.)

Resolved.

> The spare space on the left and right is no longer equally distributed:

I'm not sure I'm seeing what you mean. I did notice that the width of the middle column is now off because of the smaller font. I can see where that is coming from, and can fix that. But I'm unclear on what you're getting at with this and the screenshots. I'd appreciate some clarification.

I also found some weirdness with the "Titlebar" section - it has a background color for some reason, that isn't as noticeable on dark mode, but it's black on light mode. Trying to sort out what's going on there as well.

kattni commented:
> In the old theme, the space on the left of the left sidebar (measured from the T of "Toga), is equal to the space on the right of the right sidebar (measured from its scroll bar). In the new theme, there is more space on the right. This is visible at all window widths, but is particularly noticeable when the space is small.

I pushed the fix for the content width, which I'm relatively sure resolved the sidebar situation. Please let me know if it does not.

@mhsmith Are you seeing the following issue when rendering the demo site? I am struggling to resolve it, and wanted to make sure it wasn't something on my end specifically.


![Screenshot 2025-06-24 at 16 40 24](https://github.com/user-attachments/assets/735e4843-fb77-4709-97ae-b134e98c9588)

kattni commented:
Superseded by #7. Closing.
